Northstar.Custom/mod/scripts/vscripts/sh_server_dialog.gnut

```diff
@@ -0,0 +1,296 @@
+
+    file.playerButtonCallBack[ player ][ index ]( player )
+
+    return true
+}
+
+#elseif CLIENT
+
+void function ServerDialogueOnClientInit()
+{
+    AddServerToClientStringCommandCallback( "ServerRequestRequest", ServerCallback_DialogueRequestReceived )
+}
+
+//  string header
+//  string message
+//  string image
+//  string rightImage
+//  bool forceChoice
+//  bool noChoice
+//  bool noChoiceWithNavigateBack
+//  bool showSpinner
+//  bool showPCBackButton
+//  float inputDisableTime
+//  bool darkenBackground
+//  bool useFullMessageHeigh
+
+//  array<ServerDialogButtonData> buttonData
+//  the long array is in this order
+
+void function ServerCallback_DialogueRequestReceived( array<string> args )
+{
+    if ( args.len() != 20 ) // WHY MAKE THIS NOT DYNAMIC
+        return
+    
+    string header = StringReplace( args[0], "/s", " ", true, true )
+
+    string message = StringReplace( StringReplace(args[1], "!x1ba", "\n" ,true, true), "/s", " ", true, true )
+
+    string image = args[2]
+    string rightImage = args[3]
+    bool forceChoice = args[4].tointeger() == 1
+    bool noChoice = args[5].tointeger()  == 1
+    bool noChoiceWithNavigateBack = args[6].tointeger() == 1
+    bool showSpinner = args[7].tointeger() == 1
+    bool showPCBackButton = args[8].tointeger() == 1
+    float inputDisableTime = args[9].tofloat()
+    bool darkenBackground = args[10].tointeger() == 1
+    bool useFullMessageHeight = args[11].tointeger() == 1
+
+    string button0Text = StringReplace( args[12], "/s", " ", true, true )
+    string button1Text = StringReplace( args[13], "/s", " ", true, true )
+    string button2Text = StringReplace( args[14], "/s", " ", true, true )
+    string button3Text = StringReplace( args[15], "/s", " ", true, true )
+
+
+    int colour_r = int(args[16])
+    int colour_g = int(args[17])
+    int colour_b = int(args[18])
+    int colour_a = int(args[19])
+
+
+    RunUIScript( 
+        "ClientCallback_RequestDialog",
+        header,
+        message,
+        image,
+        rightImage,
+        forceChoice,
+        noChoice,
+        noChoiceWithNavigateBack,
+        showSpinner,
+        showPCBackButton,
+        inputDisableTime,
+        darkenBackground,
+        useFullMessageHeight,
+        button0Text,
+        button1Text,
+        button2Text,
+        button3Text,
+        colour_r,
+        colour_g,
+        colour_b,
+        colour_a
+    )
+}
+
+#elseif UI
+
+
+
+void function ClientCallback_RequestDialog( string header, string message, string image, string rightImage, bool forceChoice, bool noChoice, bool noChoiceWithNavigateBack, bool showSpinner, bool showPCBackButton, float inputDisableTime, bool darkenBackground, bool useFullMessageHeight, string button0Text, string button1Text, string button2Text, string button3Text, int colour_r, int colour_g, int colour_b, int colour_a  )
+
+
+{   
+    if ( uiGlobal.activeMenu != null )
+        return
+    
+    if ( cooldown >= Time() )
```

Comment:
Yeah, I do agree that no cool down is better 

but I did add it because of a comment on the last pr -> https://github.com/R2Northstar/NorthstarMods/pull/380#issuecomment-1213386147

so I will replace it with an optional convar for the cool down